category_theory/limits/cones.lean

```diff
@@ -0,0 +1,273 @@
+-- Copyright (c) 2017 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Stephen Morgan, Scott Morrison
+
+import category_theory.natural_isomorphism
+import category_theory.whiskering
+import category_theory.discrete_category
+import category_theory.const
+
+universes u u' v
+
+open category_theory
+
+variables (J : Type v) [small_category J]
+variables (C : Type u) [𝒞 : category.{u v} C]
+include 𝒞
+
+variables {J C}
+open category_theory
+open category_theory.functor
+
+namespace category_theory.limits
+
+def functor.of_obj (X : C) : punit.{v+1} ⥤ C :=
```

Comment:
this is now in the namespace `category_theory.limits.functor.of_obj`

I think the best is to move it to `category_theory.functor`

Reply:
I moved this to `discrete_category.lean` for now, as it relies on the category structure for `punit`, which is introduced there. Do you think, @johoelzl, I should move both of those back to `functor.lean`? It seems reasonable.

In any case, it's now in the correct namespace.